src/realm/impl/transact_log.hpp

```diff
@@ -998,6 +1062,24 @@ void TransactLogParser::parse_one(InstructionHandler& handler)
                 parser_error();
             return;
         }
+        case instr_SetInsert: {
+            size_t set_ndx = read_int<size_t>(); // Throws
+            if (!handler.set_insert(set_ndx))    // Throws
+                parser_error();
+            return;
+        }
+        case instr_SetErase: {
+            size_t set_ndx = read_int<size_t>(); // Throws
+            if (!handler.set_erase(set_ndx))     // Throws
+                parser_error();
+            return;
+        }
+        case instr_SetClear: {
```

Comment:
`SetClear` and `ArrayClear` are "weird" because they override anything that isn't causally related, regardless of timestamp, but that's sometimes the right thing to do - for instance, when assigning an array/set to a field where you want to treat that assignment as a single operation.

Of course, SDKs can do what they want. :-)